source/patterns/@aws-solutions-constructs/aws-iot-sqs/test/iot-sqs.test.ts

```diff
@@ -0,0 +1,481 @@
+// Testing with existing KMS key
+// --------------------------------------------------------------
+test('Pattern deployment with existing KMS key', () => {
+  // Initial Setup
+  const stack = new Stack();
+
+  const kmsKey = new kms.Key(stack, 'existing-key', {
+    enableKeyRotation: false,
+    alias: 'existing-key-alias'
+  });
+
+  const props: IotToSqsProps = {
+    encryptionKey: kmsKey,
+    iotTopicRuleProps: {
+      topicRulePayload: {
+        ruleDisabled: false,
+        description: "Processing messages from IoT devices or factory machines",
+        sql: "SELECT * FROM 'test/topic/#'",
+        actions: []
+      }
+    }
+  };
+  new IotToSqs(stack, 'test-iot-sqs', props);
+
+  expect(SynthUtils.toCloudFormation(stack)).toMatchSnapshot();
+
+  // Creates a default sqs queue
+  expect(stack).toHaveResource("AWS::SQS::Queue", {
+    KmsMasterKeyId: {
+      "Fn::GetAtt": [
+        "existingkey205DFC01",
+        "Arn"
+      ]
+    }
+  });
+
+  // Creates a dead letter queue
+  expect(stack).toHaveResource("AWS::SQS::Queue", {
+    KmsMasterKeyId: "alias/aws/sqs"
+  });
+
+  // Creates an IoT Topic Rule
+  expect(stack).toHaveResource("AWS::IoT::TopicRule", {
+    TopicRulePayload: {
+      Actions: [
+        {
+          Sqs: {
+            QueueUrl: { Ref: "testiotsqsqueue630B4C1F" },
+            RoleArn: {
+              "Fn::GetAtt": [
+                "testiotsqsiotactionsrole93B1D327",
+                "Arn"
+              ]
+            }
+          }
+        }
+      ],
+      Description: "Processing messages from IoT devices or factory machines",
+      RuleDisabled: false,
+      Sql: "SELECT * FROM 'test/topic/#'"
+    }
+  });
+
+  // Uses the provided key
+  expect(stack).toHaveResource("AWS::KMS::Key", {
+    EnableKeyRotation: false
+  });
+});
+
+// --------------------------------------------------------------
+// Testing with passing KMS key props
+// --------------------------------------------------------------
+test('Pattern deployment passing KMS key props', () => {
+  // Initial Setup
+  const stack = new Stack();
+
+  const props: IotToSqsProps = {
+    encryptionKeyProps: {
+      enableKeyRotation: false,
+      alias: 'new-key-alias-from-props'
+    },
+    iotTopicRuleProps: {
+      topicRulePayload: {
+        ruleDisabled: false,
+        description: "Processing messages from IoT devices or factory machines",
+        sql: "SELECT * FROM 'test/topic/#'",
+        actions: []
+      }
+    }
+  };
+  new IotToSqs(stack, 'test-iot-sqs', props);
+
+  expect(SynthUtils.toCloudFormation(stack)).toMatchSnapshot();
+
+  // Creates a default sqs queue
+  expect(stack).toHaveResource("AWS::SQS::Queue", {
+    KmsMasterKeyId: {
+      "Fn::GetAtt": [
+        "testiotsqsEncryptionKey64EE64B1",
+        "Arn"
+      ]
+    }
+  });
+
+  // Creates a dead letter queue
+  expect(stack).toHaveResource("AWS::SQS::Queue", {
+    KmsMasterKeyId: "alias/aws/sqs"
```

Comment:
???
This is the check you use when enableEncryptionWithCustomerManagedKey is false. This test is supply keyProps - shouldn't you be looking for the key alias in those props? And, if so, why does this test pass?

Reply:
If you pass `enableEncryptionWithCustomerManagedKey=false`, you are indicating that you don't want to use a customer managed key but the queue will still have the default encryption: AWS managed CMK for SQS (the alias for which is `alias/aws/sqs`). So that's why this test passes.

The test for `enableEncryptionWithCustomerManagedKey=false` similarly tests that the queues have the default SQS key but also [specifically checks](https://github.com/awslabs/aws-solutions-constructs/pull/267/files?file-filters%5B%5D=.json&file-filters%5B%5D=.png&file-filters%5B%5D=.ts#diff-dddf8da355517dd9cde214c10df35247ca9d9c28f4917c6b5d7b9f194020bc54R298-R299) that a Key was not created